ngsderive/utils.py

```diff
@@ -117,17 +115,14 @@ def __next__(self):
                     "read_group": read_group,
                     "quality": quality,
                 }
-            else:
```

Comment:
This could almost certainly be nicer: I'd define a base result with the keys you expect every time (`query_name` and `query`) and then conditionally add the `read_group` and `quality` keys to the result if the respect booleans are true.

Reply:
I'll take another look at this section 👍